vmm/sandbox/src/stratovirt/factory.rs

```diff
@@ -147,8 +153,10 @@ impl VMFactory for StratoVirtVMFactory {
             share_fs_path.as_str(),
         );
 
-        // set pcie-root-ports for hotplugging
-        vm.create_pcie_root_ports(PCIE_ROOTPORT_CAPACITY)?;
+        if vm.config.machine.r#type.split(',').next().unwrap() != MACHINE_TYPE_MICROVM {
```

Comment:
Be careful with `unwrap()`. How about `collect()` substrings after `split()`, and handle the string array.

Reply:
fixed